### PR TITLE
fix for redirect issue

### PR DESCRIPTION
**Pull request by thehesiod**
resolves https://github.com/aio-libs/aiohttp/issues/2022

- fixes Makefile to run when you have py2 + py3 installed
- removes ClientRedirectError
- ~~ran isort based on recommendations from make test~~ gave different results

### CODE REVIEW

**Comment by thehesiod** [comment]
@asvetlov  unfortunately when running the tests locally I'm getting "ValueError: option names {'--fast'} already added" so this may take a few iterations.  Let me know what you think.

**Comment by thehesiod** [comment]
ah figured it out, I had pytest-aiohttp installed, perhaps should detect/try uninstalling this first before running tests

**Comment by thehesiod** [comment]
another strange thing is running isort locally gave me a different result for background_tasks.py than through appveyor.  Perhaps needs to be pinned?

File: Makefile [hunk 1]
```diff
@@ -1,9 +1,11 @@
 # Some simple testing tasks (sorry, UNIX only).
 
+pytest := python3 -m pytest
```
Comment (asvetlov) [inline Makefile]: What is intention for this variable? It's basically an alias for just `pytest`, isn't it?

File: Makefile [hunk 2]
```diff
@@ -1,9 +1,11 @@
...
+
 all: test
 
 .install-deps: requirements/dev.txt
-	@pip install -U -r requirements/dev.txt
+	@pip3 install -U -r requirements/dev.txt
```
Comment (asvetlov) [inline Makefile]: aiohttp development assumes virtual environment usage.
In proper virtual env `pip` is just an alias for `pip3` if python is python3

File: demos/polls/aiohttpdemo_polls/db.py [hunk 3]
```diff
@@ -1,6 +1,7 @@
-import aiopg.sa
 import sqlalchemy as sa
 
+import aiopg.sa
```
Comment (asvetlov) [inline demos/polls/aiohttpdemo_polls/db.py]: Please drop this file from PR. The change is not related to issue.

File: tox.ini [hunk 4]
```diff
@@ -43,7 +43,7 @@ deps =
 
 commands =
     flake8 aiohttp examples tests
-    python setup.py check -rm
+    python3 setup.py check -rm
```
Comment (asvetlov) [inline tox.ini]: Again we are in python3 virtual environment here.
Don't need `python3` command.

File: Makefile (hunk 1 above)
Comment (thehesiod) [reply Makefile]: yes, this is so I could run the tests locally, where I have py2 + py3 installed

File: demos/polls/aiohttpdemo_polls/db.py (hunk 3 above)
Comment (thehesiod) [reply demos/polls/aiohttpdemo_polls/db.py]: when I run the tests locally flake said to run isort against this file, still want me to revert?

File: Makefile (hunk 1 above)
Comment (thehesiod) [reply Makefile]: yes, with this change you can run the tests outside virtualenv, still want me revert?  it makes no different inside virtual env but fixes it outside virtual env

**Comment by asvetlov** [comment]
Well, your changes about `python3` and `pip3` make sense maybe (but we encourage virtualenv usage in our doc for developers: http://aiohttp.readthedocs.io/en/stable/contributing.html

But I'm striving to have relative small PRs for solving one issue per PR.
If you want apply `pip3` changes etc -- let's create another PR but keep this one minimalistic.

**Comment by asvetlov** [comment]
Last thing: drop `changes/2009.feature` and add a new changelog file please.

**Comment by thehesiod** [comment]
added, btw should adding files to changes/ be documented in CONTRIBUTING.rst?

**Comment by asvetlov** [comment]
No, sorry.
But it is mentioned in GitHub Pull Request template `.githib/PULL_REQUEST_TEMPLATE.md`.

I very ask you to make a PR for CONTRIBUTING.rst update.

File: changes/2030.feature [hunk 5]
```diff
@@ -0,0 +1,3 @@
+ClientRedirectError is no longer raised
```
Comment (asvetlov) [inline changes/2030.feature]: As we don't publish aiohttp release with `ClientRedirectError` the name should be changed to `RuntimeError` (used to be raised in aiohttp 2.2)

**Comment by asvetlov** [comment]
Thanks for your patience :)

**Comment by asvetlov** [comment]
@thehesiod ` CONTRIBUTING.rst` updated

**Comment by thehesiod** [comment]
@asvetlov no worries, thank you too for the review!
